snuba/clusters/cluster.py

```diff
@@ -95,13 +119,27 @@ def get_writer(
             table_name, self.__host, self.__http_port, encoder, options, chunk_size
         )
 
+    def get_nodes(self) -> Sequence[ClickhouseHost]:
+        if self.__is_distributed:
+            # Get the nodes from system.clusters
+            return [
+                ClickhouseHost(*host)
+                for host in self.get_clickhouse_ro().execute(
+                    f"select host_name, port, shard_num, replica_num from system.clusters where cluster={escape_string(self.__cluster_name)}"
+                )
+            ]
+        else:
+            return [ClickhouseHost(self.__host, self.__port)]
```

Comment:
Why do we need to instantiate the cluster passing the `distributed` flag? Can't we infer whether we are distributed mode or not from the clickhouse response to the query you run on system.clusters ?

Reply:
The `system.clusters` table is not reliable when distributed=False, since there is a bunch of boilerplate in there. The distributed flag in this case isn't actually particularly useful and could be dropped. We can infer it from whether a user passes in a cluster_name, but I added it to be a bit more explicit since it might be easy for the user to inadvertently omit the cluster name. The "cluster_name" is the one that actually provides the name that is looked up in the system.clusters table.

Reply:
ok